packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,502 @@
+      class="timepicker-input"
+    >
+      {{ abbreviatedDisplay }}
+    </KButton>
+    <template
+      v-if="!hidePopover"
+      #content>
+      <KSegmentedControl
+        v-if="hasRelativeTimeframes && hasCalendar"
+        v-model="tabName"
+        :options="[
+          { label: 'Relative', value: 'relative' },
+          { label: 'Custom', value: 'custom' }
+        ]"
+        class="w-100 mb-5"
+        data-testid="analytics-time-toggle"
+      />
+      <p v-if="!showCalendar">{{ fullRangeDisplay }}</p>
+      <DatePicker
+        v-if="hasCalendar && showCalendar"
+        v-model="selectedCalendarRange"
+        :is-range="isRange"
+        :select-attribute="calendarSelectAttributes"
+        :model-config="modelConfig"
+        :mode="calendarMode"
+        is-expanded
+      />
+      <div
+        v-else-if="hasRelativeTimeframes && hasTimePeriods"
+        class="d-flex flex-column"
+      >
+        <div
+          v-for="(item, index) in timePeriods"
+          :key="`section-${index}`"
+          class="timeframe-section d-flex flex-column"
+        >
+          <h4>{{ item.section }}</h4>
+          <div class="timeframe-buttons d-flex flex-row justify-content-start">
+            <KButton
+              v-for="(timeFrame, index) in item.values"
+              :key="`time-${index}`"
+              :is-rounded="false"
+              :class="{'selected-option': timeFrame.timeframeText === selectedTimeframe.timeframeText}"
+              :data-testid="'select-timeframe-' + timeFrame.timeframeLength()"
+              appearance="outline"
+              size="medium"
+              @click="changeRelativeTimeframe(timeFrame)"
+            >
+              {{ timeFrame.timeframeText }}
+            </KButton>
+          </div>
+        </div>
+      </div>
+    </template>
+    <template
+      v-if="!hidePopover"
+      #footer>
+      <div class="d-flex justify-content-end">
+        <KButton
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="clearSelection()"
+        >
+          Clear
+        </KButton>
+        <KButton
+          :disabled="submitDisabled"
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="submitTimeFrame()"
+        >
+          Apply
+        </KButton>
+      </div>
+    </template>
+  </KPop>
+</template>
+
+<script>
+import KButton from '@kongponents/kbutton/KButton.vue'
+import KPop from '@kongponents/kpop/KPop.vue'
+import KSegmentedControl from '@kongponents/ksegmentedcontrol/KSegmentedControl.vue'
+import { format } from 'date-fns'
+
+export default {
+  name: 'KTimePicker',
+  components: {
+    KButton,
+    KPop,
+    KSegmentedControl,
+    DatePicker: () => import('v-calendar/lib/components/date-picker.umd')
+  },
+  props: {
+    defaultMessage: {
+      type: String,
+      required: false,
+      default: 'Select a time range'
+    },
+    calendarMode: {
```

Comment:
You'll need to update the usage throughout the component as well.
```suggestion
    mode: {
```